flink-java/src/main/java/org/apache/flink/api/java/sampling/PoissonSampler.java

```diff
@@ -93,29 +98,59 @@ public boolean hasNext() {
 					}
 				}
 			}
-
-			private void moveToNextElement() {
-				while (input.hasNext()) {
+			
+			public int poisson_ge1(double p){
+				// sample 'k' from Poisson(p), conditioned to k >= 1
+				double q = Math.pow(Math.E, -p);
+				// simulate a poisson trial such that k >= 1
+				double t = q + (1 - q)*random.nextDouble();
+				int k = 1;
+				// continue standard poisson generation trials
+				t = t * random.nextDouble();
+				while (t > q) {
+					k++;
+					t = t * random.nextDouble();
+				}
+				return k;
+			}
+			
+			private void moveToNextElement(int num) {
```

Comment:
This method skip gap elements, you actually move to the next sampled element in ```samplingProcess```, how about rename to ```skipGapElements```?